.github/workflows/docs-pull-request.yaml

```
@@ -0,0 +1,53 @@
+# This workflow builds API docs on pull requests that touch the parts
+# of the api/ project that are used by the docs; the docs source themselves;
+# or the CI/makefile tooling used to build the docs.
+
+name: 'API docs'
```

To keep things consistent:
```suggestion
name: 'API docs pull requests'
```